src/mame/drivers/mc10.cpp

```diff
@@ -167,47 +158,47 @@ READ8_MEMBER( mc10_state::mc10_port2_r )
 {
 	uint8_t result = 0xeb;
 
-	/* bit 1, keyboard line pa6 */
-	if (!BIT(m_keyboard_strobe, 0)) result &= ioport("pb0")->read() >> 5;
-	if (!BIT(m_keyboard_strobe, 2)) result &= ioport("pb2")->read() >> 5;
-	if (!BIT(m_keyboard_strobe, 7)) result &= ioport("pb7")->read() >> 5;
+	// bit 1, keyboard line pa6
+	if (!BIT(m_keyboard_strobe, 0)) result &= m_pb[0]->read() >> 5;
+	if (!BIT(m_keyboard_strobe, 2)) result &= m_pb[2]->read() >> 5;
+	if (!BIT(m_keyboard_strobe, 7)) result &= m_pb[7]->read() >> 5;
 
-	/* bit 2, printer ots input */
+	// bit 2, printer ots input
 	result |= (m_printer->is_ready() ? 0 : 4);
 
-	/* bit 3, rs232 input */
+	// bit 3, rs232 input
 
-	/* bit 4, cassette input */
+	// bit 4, cassette input
 	result |= ((m_cassette)->input() >= 0 ? 1 : 0) << 4;
```

Comment:
This is not consistent with the one above.  Above you've got `cond ? 0 : n` while here you've got `(cond ? 1 : 0) << 4`.  Make both of them use 0/1 and a shift, or make neither of them use a shift.